packages/conf-aarch64-linux-gnu-gcc/conf-aarch64-linux-gnu-gcc.1/opam

```diff
@@ -0,0 +1,20 @@
+opam-version: "2.0"
+synopsis: "Virtual package relying on the aarch64-linux-gnu-gcc compiler (for C)"
+description:
+  "This package can only install if the aarch64-linux-gnu-gcc compiler is installed on the system (whether this is a cross compiler or not)."
+maintainer: "samuel@tarides.com"
+authors: ["Samuel Hym"]
+license: "GPL-2.0-or-later"
+homepage: "https://github.com/ocaml/opam-repository"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+flags: conf
+build: ["aarch64-linux-gnu-gcc" "--version"]
+depexts: [
+  ["gcc-aarch64-linux-gnu"] {os-family = "debian"}
+  ["gcc-aarch64-linux-gnu"] {os-family = "fedora"}
+  ["aarch64-linux-gnu-gcc"] {os-family = "arch"}
+  ["no-known-package"]
+    {os-distribution = "alpine" | os-family = "suse" | os-family = "opensuse" |
+     os-family = "bsd" | os = "macos" | os = "cygwin" | os = "win32"}
```

Comment:
I think the cleanest, and a more helpful way of addressing this would be to use a [`post-message`](https://opam.ocaml.org/doc/Manual.html#opamfield-post-messages).

Here is an example: https://github.com/ocaml/opam-repository/blob/d4d1aa63ab7830b868cb8cbd85e076c699d8eb06/packages/conf-llvm/conf-llvm.3.4/opam#L15-L17

Do you think it would make sense to use something like that in this case?

Reply:
The more uniform these cases are handled, the better!
I suppose I’d better add a post-message just on `{failure}` to cover all cases.
Do you mean I should drop the `"no-known-package"` hack also? (It’s helpful for CI but might be annoying for end users).

Reply:
> The more uniform these cases are handled, the better!
I suppose I’d better add a post-message just on {failure} to cover all cases.

Yep, this makes sense to me! Just a general note on failure to advise people why/that automated system install is not available for them would help.

>  Do you mean I should drop the "no-known-package" hack also? (It’s helpful for CI but might be annoying for end users).

Yea, IIUC, it should not be needed for our CI or for end users. But if you need it for your dev CI, I don't want to make things more complicated! Leaving is not a problem, IMO. But I would defer to @mseri or @raphael-proust on this point.

Reply:
As I was thinking about opam-CI, I updated as discussed, namely adding a `post-messages` field and removing the `"no-known-package"` hack.
(Many CI tests which were converted into `SKIP` thanks to the hack are back to failures.)

Reply:
> Many CI tests which were converted into SKIP thanks to the hack are back to failures

We can address these via the appropriate [`x-ci-accept-failures`](https://github.com/ocaml/opam-repository/wiki/How-to-deal-with-CI#annex) in the future. But I don't think this is necessary currently!

Reply:
> the check for this field doesn’t trigger as expected at least in some configurations.

Yes, I think that's true. The logic around this is not very robust.